### PR TITLE
Issue 3045: Reduce time taken for travis build by not running integration tests

### LINKED ISSUE #3045: Reduce time taken for travis builds - DONOTMERGE
**Problem description**
Travis builds run integration tests for each of the tier 2 bindings. This is time consuming. We can have substantial reduction in time taken for travis verification if we could avoid checking for `bindings` project.

**Problem location**
`bindings` build.

**Suggestions for an improvement**
Run travis with ignoring bindings project.

### PR DESCRIPTION
**Pull request by arvindkandhare**
**Change log description**  
Refactor integration tests based on tier 2 bindings
Block integration tests from running in travis.

**Purpose of the change**  
Fixes #3045.

**What the code does**  
1. Move tier 2 based tests to bindings project.
2. Introduce independent gradle file for bindings
3. Block intgration tests in travis 

**How to verify it**  
Travis build passes in less time than earlier.

### CODE REVIEW

**Comment by andreipaduroiu** [comment]
I see about 35 minutes for this PR's build vs 49 minutes for master.

File: .travis.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
     - stage: build
       if: type IN (push, pull_request)
       install: ./gradlew assemble
-      script: ./gradlew check
+      script: ./gradlew -x :bindings:check -x :standalone:check check
```
Comment (andreipaduroiu) [inline .travis.yml]: It just occurred to me that this may not be what we want. If we don't run `check` on `bindings` or `standalone` then we could end up with code degradation or even uncompilable code.

Are these built and checkstyled at least? We may not want the tests to run, but it should still be compiled.

Comment (fpj) [reply .travis.yml]: I think the intuition is right that we don't need to run the tests for bindings upon every single PR build, specially if there is no change in the bindings. The interface abstraction between the segment store and tier 2 should be sufficient to validate changes elsewhere without running all tier 2 tests.

What's bad about this line change:
- If we make changes to tier 2 bindings, then we won't be running tier 2 tests
- We also need a consistent way to test standalone, and I believe changes across pravega can affect the behavior of standalone. We could have a daily standalone build, but we would still have the same issue that changes focused on standalone (which happen rarely) would not trigger standalone tests with this definition.

File: bindings/src/test/java/io/pravega/server/host/ExtendedS3IntegrationTest.java [hunk 2]
```diff
@@ -117,7 +117,7 @@ public Storage createStorageAdapter() {
                     .withInitialRetryDelay(1)
                     .withProperty("com.sun.jersey.client.property.connectTimeout", 100);
 
-            S3JerseyClient client = new S3ClientWrapper(s3Config, filesystemS3);
+            S3JerseyClient client = new io.pravega.segmentstore.server.host.S3ClientWrapper(s3Config, filesystemS3);
```
Comment (fpj) [inline bindings/src/test/java/io/pravega/server/host/ExtendedS3IntegrationTest.java]: Isn't this class being also moved under the same package, is this change necessary?

File: build.gradle [hunk 3]
```diff
@@ -84,6 +84,7 @@ allprojects {
     }
 }
 
+
```
Comment (fpj) [inline build.gradle]: Unnecessary change.

File: settings.gradle [hunk 4]
```diff
@@ -28,3 +28,4 @@ include 'common',
         'test:integration',
         'test:system',
         'bindings'
+
```
Comment (fpj) [inline settings.gradle]: Unnecessary change.

File: bindings/build.gradle [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+def withoutLogger = { exclude group: 'org.slf4j', module: 'slf4j-log4j12'
+    exclude group: 'org.slf4j', module: 'slf4j-simple' }
+
+    dependencies {
```
Comment (fpj) [inline bindings/build.gradle]: The indentation seems incorrect here.

File: test/integration/src/main/java/io/pravega/test/integration/selftest/adapters/OutOfProcessAdapter.java [hunk 6]
```diff
@@ -226,8 +225,7 @@ private Process startSegmentStore(int segmentStoreId) throws IOException {
                 .sysProp(configProperty(ServiceConfig.COMPONENT_CODE, ServiceConfig.CERT_FILE), "../../config/cert.pem")
                 .sysProp(configProperty(BookKeeperConfig.COMPONENT_CODE, BookKeeperConfig.ZK_ADDRESS), getZkUrl())
                 .sysProp(configProperty(ServiceConfig.COMPONENT_CODE, ServiceConfig.LISTENING_PORT), port)
-                .sysProp(configProperty(ServiceConfig.COMPONENT_CODE, ServiceConfig.STORAGE_IMPLEMENTATION), ServiceConfig.StorageType.FILESYSTEM)
-                .sysProp(configProperty(FileSystemStorageConfig.COMPONENT_CODE, FileSystemStorageConfig.ROOT), getSegmentStoreStoragePath())
+                .sysProp(configProperty(ServiceConfig.COMPONENT_CODE, ServiceConfig.STORAGE_IMPLEMENTATION), ServiceConfig.StorageType.INMEMORY)
```
Comment (fpj) [inline test/integration/src/main/java/io/pravega/test/integration/selftest/adapters/OutOfProcessAdapter.java]: This is a change of behavior. Why are you suggesting we change this on selftest?

File: bindings/build.gradle (hunk 5 above)
Comment (fpj) [inline bindings/build.gradle]: This change has nothing to do with reducing the time of tests, you are trying to bring the dependencies exclusive to bindings closer to them, is that correct?

File: .travis.yml (hunk 1 above)
Comment (arvindkandhare) [reply .travis.yml]: These changes are limited to travis. All the other places where we run periodic (daily) tests will not be affected by this change.

File: bindings/build.gradle (hunk 5 above)
Comment (arvindkandhare) [reply bindings/build.gradle]: Yes. This is correct. Now that we are separating gradle files for bindings and core, looked like a right time to make the changes.

File: test/integration/src/main/java/io/pravega/test/integration/selftest/adapters/OutOfProcessAdapter.java (hunk 6 above)
Comment (arvindkandhare) [reply test/integration/src/main/java/io/pravega/test/integration/selftest/adapters/OutOfProcessAdapter.java]: As we are not building the bindings, the implementation for `FILESYSTEM` will not be loaded in travis. When this process is run, whether we write to local filesystem OR to local mem does not make any difference as both are not real deployments.

File: bindings/build.gradle (hunk 5 above)
Comment (arvindkandhare) [reply bindings/build.gradle]: Fixed.

File: settings.gradle (hunk 4 above)
Comment (arvindkandhare) [reply settings.gradle]: Fixed.

File: build.gradle (hunk 3 above)
Comment (arvindkandhare) [reply build.gradle]: Fixed.

File: bindings/src/test/java/io/pravega/server/host/ExtendedS3IntegrationTest.java (hunk 2 above)
Comment (arvindkandhare) [reply bindings/src/test/java/io/pravega/server/host/ExtendedS3IntegrationTest.java]: This is not necessary. Fixed.

**Comment by fpj** [comment]
If there is any interest in bringing this PR in, then please reopen. Closing for now.
